packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -199,11 +201,24 @@ export const prepareVaultManagerKit = (
     makeVaultManagerPublishKit();
   pipeTopicToStorage(metricsSubscriber, metricsNode, marshaller);
 
+  const quoteNotifier = E(priceAuthority).makeQuoteNotifier(
+    collateralUnit,
+    debtBrand,
+  );
   const storedQuotesNotifier = makeStoredNotifier(
-    E(priceAuthority).makeQuoteNotifier(collateralUnit, debtBrand),
+    quoteNotifier,
     E(storageNode).makeChildNode('quotes'),
     marshaller,
   );
+  let storedCollateralQuote;
+  observeNotifier(quoteNotifier, {
```

Comment:
This `observeNotifier` will fail upgrade. https://github.com/Agoric/agoric-sdk/issues/6791 the ticket to make this contract upgradable, so this PR doesn't have to be upgradable but we should at least have a sketch of how this will be upgraded.

Also it will consume heap so I want to hear from @mhofman on whether landing this would complicate his research.

Reply:
Would this result a per-vault heap object? Or does this happen only after some trigger for a vault?

We should definitely hunt down any patterns that result in high cardinality heap objects.

Reply:
Per-manager heap object. There are order 10 managers.

Reply:
Sounds like we don't need to worry about the heap usage. The upgrade aspect I'll taclke in #6791.

If you do another push, consider moving this `observeNotifier` call to `finish()` with the other one.